skimage/filters/ridges.py

```diff
@@ -106,8 +108,278 @@ def _check_sigmas(sigmas):
+            for ii in range(len(axes)) if sigmas[ii] > 1e-15]
+    if len(axes) > 0:
+        for axis, sigma, order, mode in axes:
+            gaussian_filter_with_fft_kernel(input, sigma, axis, order,
+                                            output, mode, cval, truncate)
+            input = output
+    else:
+        output[...] = input[...]
+    return output
+
+
+def gaussian_filter_with_fft_kernel(input, sigma, axis=-1, order=0,
+                                    output=None, mode="reflect", cval=0.0,
+                                    truncate=4.0):
+    """1-D Gaussian filter using a kernel computed in the freq. domain
+    Parameters
+    ----------
+    %(input)s
+    sigma : scalar
+        standard deviation for Gaussian kernel
+    %(axis)s
+    order : int, optional
+        An order of 0 corresponds to convolution with a Gaussian
+        kernel. A positive order corresponds to convolution with
+        that derivative of a Gaussian.
+    %(output)s
+    %(mode_reflect)s
+    %(cval)s
+    truncate : float, optional
+        Truncate the filter at this many standard deviations.
+        Default is 4.0.
+    Returns
+    -------
+    gaussian_filter1d : ndarray
+    Examples
+    --------
+    >>> from skimage.filters.ridges import gaussian_filter_with_fft_kernel
+    >>> from scipy.ndimage import gaussian_filter1d
+    >>> gaussian_filter_with_fft_kernel([1.0, 2.0, 3.0, 4.0, 5.0], 1)
+    array([ 1.42703014,  2.06782964,  3.        ,  3.93217036,  4.57296986])
+    >>> gaussian_filter_with_fft_kernel([1.0, 2.0, 3.0, 4.0, 5.0], 4)
+    array([ 2.91954198,  2.95027135,  3.        ,  3.04972865,  3.08045802])
+    >>> import matplotlib.pyplot as plt
+    >>> rng = np.random.default_rng()
+    >>> x = rng.standard_normal(101).cumsum()
+    >>> y3 = gaussian_filter_with_fft_kernel(x, 0.7)
+    >>> y6 = gaussian_filter_with_fft_kernel(x, 6)
+    >>> plt.plot(x, 'k', label='original data')
+    >>> plt.plot(y3, '--', label='filtered, sigma=0.7')
+    >>> plt.plot(y6, ':', label='filtered, sigma=6')
+    >>> plt.legend()
+    >>> plt.grid()
+    >>> plt.show()
+    """
+    sd = float(sigma)
+    # make the radius of the filter equal to truncate standard deviations
+    lw = int(truncate * sd + 0.5)
+    # Wavenumber domain Gaussian
+    kx = 2 * np.pi * np.fft.fftfreq(2 * lw + 1)
+    gauss_wavenumber_domain = np.exp(-kx ** 2 * sd ** 2 / 2) \
+                              * np.exp(-1j * lw * kx)
+    # Differentiate in the frequency domain (=multiplication with i*k)
+    gauss_wavenumber_domain = gauss_wavenumber_domain * (1j * kx)**order
+    # Transform to spatial domain
+    weights = np.real(np.fft.ifft(gauss_wavenumber_domain))
+    # Since we are calling correlate, not convolve, revert the kernel
+    weights = weights[::-1]
+    return ndi.correlate1d(input, weights, axis, output, mode, cval, 0)
+
+
+def hessian_matrix_with_Gaussian(image, sigma=1, mode='reflect', cval=0,
```

Comment:
our existing `hessian_matrix` works for 3D, etc. as well while this one is 2D only. This should be possible to extend to nD, though, and we can help with that.

Reply:
It works on 3D arrays (or higher) too! I copied the help text quite verbatim from the existing `hessian_matrix` code which suggests 2D use only, but I tested it on 3D arrays too (where it outputs 6 values corresponding to d_x^2, d_xd_y, d_xd_z, d_y^2, d_yd_z, d_z^2, just as the existing `hessian_matrix` code).

Reply:
Okay, great! I had not looked closely yet and had thought it was 2D only. 

By chance, I had just made a PR (merged yesterday) that updated the `hessian_matrix` docstring to make it clearer nD is supported: https://github.com/scikit-image/scikit-image/pull/5569/files